### PR TITLE
Fix Crafting Pattern Models

### PR DESCRIPTION
**Pull request by pauljoda**
This will standardize the transforms to look like a normal item and will also apply the IItemColor object from the stored item to the item, allowing things like potions, spawn eggs, and other implementations to show correct color

### CODE REVIEW

File: src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java [hunk 1]
```diff
@@ -257,6 +263,26 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
...
+    public void init(FMLInitializationEvent e) {
+        super.init(e);
+
+        // Register IItemColor to handle passthrough
+        Minecraft.getMinecraft().getItemColors().registerItemColorHandler(new IItemColor() {
+            @Override
+            public int getColorFromItemstack(ItemStack stack, int tintIndex) {
+                CraftingPattern pattern = ItemPattern.getPatternFromCache(Minecraft.getMinecraft().world, stack);
+
+                if (GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1 &&
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java]: Can we move the `GuiBase.isShiftKeyDown() && pattern.isValid() && pattern.getOutputs().size() == 1` check to a static method in `BakedModelPattern` and reuse it there as well?

**Comment by raoulvdberge** [comment]
Thanks, left one small remark.

File: src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java [hunk 2]
```diff
@@ -257,6 +263,26 @@ public void onResourceManagerReload(IResourceManager resourceManager) {
...
+                        Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex) != -1) {
```
Comment (raoulvdberge) [inline src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java]: `Minecraft.getMinecraft().getItemColors().getColorFromItemstack(pattern.getOutputs().get(0), tintIndex)` can probably go in another variable, we're reusing it twice.

Comment (pauljoda) [inline src/main/java/com/raoulvdberge/refinedstorage/proxy/ProxyClient.java]: Addressed in last commit

**Comment by raoulvdberge** [comment]
Thank you, merged.

**Pull request merged**